developers/contracts/meta-peg-in-endpoint.md

```diff
@@ -0,0 +1,319 @@
+
+#### `pause`
+A public function, governed through the `is-dao-or-extension`, that can change the contract's operational status.
+
+##### Parameters
+```lisp
+(new-paused bool)
+```
+
+#### `transfer-all-to-many`
+This contract feature is designed to transfer its entire balance of a specified list of tokens to a new owner. Access to this feature is restricted by the `is-dao-or-extension` function.
+
+##### Parameters
+```lisp
+(new-owner principal) (token-traits (list 10 <ft-trait>))
+```
+
+#### `set-fee-to-address`
+This contract feature sets the address to which the fee in Bridged BTC tokens will be transferred.
+
+##### Parameters
+```lisp
+(new-fee-to-address principal)
+```
+
+#### `set-peg-in-fee`
+This feature establishes the fee to be deducted for the peg-in operation.
+
+##### Parameters
+```lisp
+(fee uint)
+```
+
+#### `set-btc-peg-out-fee`
+###### _(only present in meta-peg-in-v2-06-swap)_
+This feature sets the percentage fee applied to BTC peg-out operations.
+
+##### Parameters
+```lisp
+(fee uint)
+```
+
+#### `set-btc-peg-out-min-fee`
+###### _(only present in meta-peg-in-v2-06-swap)_
+This feature establishes the minimum fee required for BTC peg-out operations.
+
+##### Parameters
+```lisp
+(fee uint)
+```
+
+### Getters
+
+* `get-fee-to-address`
+* `get-peg-in-fee`
+* `get-pair-details`
+* `get-pair-details-or-fail`
+* `get-pair-details-many`
+* `get-tick-to-pair-or-fail`
+* `get-peg-in-sent-or-default`
+* `get-btc-peg-out-fee` _(only present in meta-peg-in-v2-06-swap)_
+* `get-btc-peg-out-min-fee` _(only present in meta-peg-in-v2-06-swap)_
+* `get-liabtc-decimals` _(only present in meta-peg-in-endpoint-v2-04-lisa)_
+
+### Relevant internal functions
+
+* `index-tx`: this function validates and indexes a Bitcoin transaction in the `oracle-v2-01` contract.
+* `refund`: this function returns the total amount of the peg-in transaction, including the fee, to the sender in case of a failure during the operation.
+
+## Contract calls (interactions)
+
+* `executor-dao`: calls are made to verify whether a certain contract-caller is designated as an extension.
+* `meta-bridge-registry-v2-03`: this interaction is used by the peg-in contract to check and set token pair information. It also facilitates liquidity operations to set balances and manage pool operations.
+* `btc-peg-out-endpoint-v2-01`: the `request-peg-out-0` function of this contract is used by the peg-in contract to remove liquidity and refund BTC amounts.
+* `btc-bridge-registry-v2-01`: the peg-in contract calls the `set-peg-in-sent` function of this contract to register a processed Bitcoin transaction.
+* `meta-peg-out-endpoint-v2-04`: the `finalize-peg-in-remove-liquidity` feature and the internal `refund` function call this contract to request a peg-out in their respective contexts.
+* `cross-router-v2-03`: the peg-in cross and peg-in cross swap operations use this contract to validate and route peg-ins involving other blockchains.
```

Comment:
from :  .. use this contract to ~~validate~~ and route..
to:  .. use this contract to **validate-route** and route..